src/Employer/Employer.Web/Controllers/Part1/TrainingProviderController.cs

```diff
@@ -20,25 +20,33 @@ public class TrainingProviderController : Controller
         private const string InvalidUkprnMessageFormat = "The UKPRN {0} is not valid or the associated provider is not active.";
         private const string InvalidSearchTerm = "Please enter a training provider name or UKPRN";
 
-        public TrainingProviderController(TrainingProviderOrchestrator orchestrator, IRecruitVacancyClient vacancyClient)
+        public TrainingProviderController(TrainingProviderOrchestrator orchestrator)
         {
             _orchestrator = orchestrator;
         }
 
         [HttpGet("select-training-provider", Name = RouteNames.TrainingProvider_Select_Get)]
         public async Task<IActionResult> SelectTrainingProvider(VacancyRouteModel vrm, [FromQuery] string wizard = "true", [FromQuery] string clear = "", [FromQuery] long? ukprn = null)
         {
-            var vm = await _orchestrator.GetSelectTrainingProviderViewModelAsync(vrm, ukprn);
-            vm.PageInfo.SetWizard(wizard);
+           var vm = await _orchestrator.GetSelectTrainingProviderViewModelAsync(vrm, ukprn);
+           vm.PageInfo.SetWizard(wizard);
+
+           if (string.IsNullOrWhiteSpace(clear) == false)
+           {
+               vm.Ukprn = string.Empty;
+               vm.TrainingProviderSearch = string.Empty;
+               vm.IsTrainingProviderSelected = true;
+           }
+           
+           vm.BackLinkRoute = ShowReferredFromMaBackLink(vrm.VacancyId) ? RouteNames.Title_Get : RouteNames.Training_Get;
+           return View(vm);
+        }
 
-            if (string.IsNullOrWhiteSpace(clear) == false)
-            {
-                vm.Ukprn = string.Empty;
-                vm.TrainingProviderSearch = string.Empty;
-                vm.IsTrainingProviderSelected = true;
-            }
-            
-            return View(vm);
+        private bool ShowReferredFromMaBackLink(Guid vacancyId)
```

Comment:
Return a string instead of bool and return the route name then you won't need to double up of IF statements.  Rename to `GetBackLinkRoute()`.